pkg/controller/appmetricsapp/appmetricsapp_controller.go

```diff
@@ -0,0 +1,204 @@
+	instance := &metricsv1alpha1.AppMetricsApp{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	is, err := IsValidAppNamespace(instance.ObjectMeta.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !is {
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	routeList := &routev1.RouteList{}
+	listOptions := &client.ListOptions{}
+	listOptions.InNamespace("app-metrics")
+	listOptions.MatchingLabels(map[string]string{"component": "aerogear-app-metrics"})
+	err = r.client.List(context.TODO(), listOptions, routeList)
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if len(routeList.Items) != 1 {
+		err = fmt.Errorf("Found more or less than one Route in the namespace for the metrics service")
+		return reconcile.Result{}, err
+	}
+
+	// Define a new ConfigMap object
+	configMap := newConfigMapForCR(instance, routeList.Items[0].Spec.Host)
+
+	// Set AppMetricsApp instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, configMap, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if this ConfigMap already exists
+	found := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: configMap.Name, Namespace: configMap.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new ConfigMap", "ConfigMap.Namespace", configMap.Namespace, "ConfigMap.Name", configMap.Name, "cr.Spec.Name", instance.Spec.Name)
+		err = r.client.Create(context.TODO(), configMap)
+		if err != nil {
+			reqLogger.Info("Error creating the new ConfigMap", "ConfigMap.Namespace", configMap.Namespace, "ConfigMap.Name", configMap.Name, "Error", err)
+			return reconcile.Result{}, err
+		}
+
+		// ConfigMap created successfully - don't requeue
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// ConfigMap already exists - don't requeue
+	reqLogger.Info("Skip reconcile: ConfigMap already exists", "ConfigMap.Namespace", found.Namespace, "ConfigMap.Name", found.Name)
+	return reconcile.Result{}, nil
+}
+
+func newConfigMapForCR(cr *metricsv1alpha1.AppMetricsApp, host string) *corev1.ConfigMap {
+	configmapname := fmt.Sprintf("%s-metrics", cr.Spec.Name)
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configmapname,
+			Namespace: "mobile-developer-console",
+		},
+		Data: map[string]string{
+			"SDKConfig": fmt.Sprintf("{\"url\": \"https://%s\"}", host),
+		},
+	}
+}
+
+func GetAppNamespaces() (string, error) {
```

Comment:
This is an oddly specific version of a more generic function that could have the signature `func getEnvVar(envVar string) (string, error)`, which may or may not exist. My guess is that it doesn't exist because it's not really necessary.

Reply:
Done